distributed/worker.py

```diff
@@ -2982,8 +2982,11 @@ def _ensure_communicating(self, *, stimulus_id: str) -> RecsInstrs:
             if ts.state != "fetch" or ts.key in all_keys_to_gather:
                 continue
 
+            if not ts.who_has:
+                recommendations[ts] = "missing"
+                continue
+
```

Comment:
Must remove assertion (and add a comment why the assertion isn't there) in validate_task_fetch